src/main/java/com/icthh/xm/ms/entity/service/impl/FunctionServiceImpl.java

```diff
@@ -68,15 +69,16 @@ public FunctionContext execute(final String functionKey, final Map<String, Objec
         dynamicPermissionCheckService.checkContextPermission(DynamicPermissionCheckService.FeatureContext.FUNCTION,
             FUNCTION_CALL_PRIV, functionKey);
 
-        FunctionSpec functionSpec = findFunctionSpec(functionKey, null);
+        FunctionSpec functionSpec = findFunctionSpec(functionKey, httpMethod);
 
         validateFunctionInput(functionSpec, functionInput);
 
         enrichInputFromPathParams(functionKey, vInput, functionSpec);
 
         // execute function
         return callLepExecutor(functionSpec.getTxType(), () -> {
-            Map<String, Object> data = functionExecutorService.execute(functionSpec.getKey(), vInput, httpMethod);
+            var lepHttpMethod = POST_URLENCODED.equals(httpMethod) ? "POST" : httpMethod;
```

Comment:
this logic is in two places. consider extracting method

Reply:
done